src/WriteCapInfo.cpp

```diff
@@ -58,8 +58,23 @@ inline int get_file_size(std::string path)
 
 map<int,pair<string,string>> cached_args;
 
+SEXP makeError(string description)
+{
+    return List::create(
+        Named("type") = Rf_mkString("error"),
+        Named("error_description") = Rf_mkString(description.c_str())
+    );
+}
+
 // [[Rcpp::export]]
-void enterFunction_cpp (CharacterVector fname, SEXP args_list, SEXP call_id) {
+void clearCallCache_cpp ()
+{
+    cached_args.clear();
+}
+
+// [[Rcpp::export]]
+SEXP enterFunction_cpp (CharacterVector fname, SEXP args_list, SEXP call_id)
```

Comment:
Is the `_cpp` some sort of a convention in Rcpp? If not how about calling it `do_enter_function`